test/system/setup_suite.bash

```diff
@@ -39,11 +39,11 @@ function teardown_suite() {
     stop_registry
     local exit_code=$?
 
-    # After all tests make sure there are no leaks and cleanup if there are
+    # After all tests, make sure there are no leaks
+    # FIXME: this should be done only if all tests pass. How can we find out?
```

Comment:
I think it would be easy to check BATS_TEST_COMPLETED in teardown and if the test failed touch a file in $BATS_SUITE_TMPDIR and then at the end check if file is there and skip the leak check if we had a failure